plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -829,8 +838,24 @@ def download_rpms(self, metadata_files, rpms_to_download, url):
 
             if self.download_deferred:
                 catalog = PackageCatalog(self.conduit.importer_object_id, url)
+                importer = Importer.objects.get(id=self.conduit.importer_object_id)
+
                 for unit in units_to_download:
                     unit.downloaded = False
+                    # If feed-url has changed since last-sync, we need to:
+                    #   - delete existing matching-unit if there is one, and
+                    #   - delete existing LazyCatalogEntry if there is one,
+                    # in order to safely rebuild LCE with possibly-new download-urls
+                    # [NOTE: "if there is one" is almost certainly paranoia at this point in the
+                    # code-flow. Paranoia is not always invalid, however...]
+                    if importer_constants.KEY_FEED_UPDATED in importer.config and \
+                            importer.config[importer_constants.KEY_FEED_UPDATED]:
+                        units = unit.__class__.objects.filter(**unit.unit_key)
+                        repo_controller.disassociate_units(self.repo, units)
```

Comment:
why do we dissociate the units here? I think if the repo is in additive mode( which is by default) then we should not remove any units. Only if it is in mirror mode ( remove_missing set to true) then we exactly mirror our local repo to a remote one. Is my thinking wrong?

Reply:
If the new feed has the same NEVRA as the old, we 'assume' the desired end-result is to be getting that content from new-feed,  even if it was available from old-feed.

Reply:
right, for that i think we just need to update the LCE to point to the new feed, it does not require disassociating the unit.

```
Repo in additive mode has X, Y, Z packages.
New feed has X, P packages. 
I expect to  1) For X update the LCE to point to new feed 2) Y,Z stay in the repo, LCE untouched 3) P new  package in the repo + LCE

Repo in mirror mode has X, Y, Z packages.
New feed has X, P packages. 

I expect to  1)  For X update the LCE to point to new feed 2) Y,Z disassociate from the repo, LCE untouched 3) P new  package in the repo + LCE
```

@dkliban  i know you participated in discussion of the implementation, can you confirm that these expectations are legit?

Reply:
@ipanova iirc, 'just update the LCE' turned out to be Fraught With Peril, for reasons I confess I don't recall - it's been a while. Among other things, dealing with "the path to the NEVRA on the new feed may be wildly different than the one leading to the same NEVRA, on the old feed', was fun trying to track down. 

IIRC, deleting and recreating the LCE was more reliable than trying to modify a/the existing one, coupled with "deleting the old one without deleting the content-unit it 'knew about' causes duplicates...somewhere". I don't remember exactly where, I'm afraid -  I'll have to try and reproduce the investigation that led us here, may take some time.

Reply:
@ggainey, @ipanova: I read through the implementation of the sync code and I don't quite understand why removing the unit is necessary.

Reply:
OK folks, I sent a separate email on how this happened. Current state does NOT delete content-units Just Because - please re-review!